mex/common/wikidata/convenience.py

```diff
@@ -0,0 +1,52 @@
+from collections.abc import Callable, Iterable
+
+from mex.common.models import ExtractedData, ExtractedPrimarySource
+from mex.common.types import MergedOrganizationIdentifier
+from mex.common.wikidata.extract import search_organization_by_label
+from mex.common.wikidata.transform import (
+    transform_wikidata_organization_to_extracted_organization,
+)
+
+_ORGANIZATION_BY_QUERY_CACHE: dict[str, MergedOrganizationIdentifier] = {}
```

Comment:
how reliable is this solution? is this better than using functools `cache` as a decorator on this function?

Reply:
`functools.cache` cannot deal with `ExtractedPrimarySource` and `Callable`.
I did not either, though. I adapted the cache to be specific to these call args in 201d7721fc9071285461a793dd44aaf6b46b0555